app/assets/javascripts/initializers/initializeAllFollowButts.js

```diff
@@ -94,7 +94,26 @@ function handleOptimisticButtRender(butt) {
     showModal('follow-button');
   } else {
     // Handles actual following of tags/users
-    assignState(butt, butt.dataset.verb);
+    try {
+        //lets try grab the event buttons info data attribute user id
+        let evFabUserId = JSON.parse(butt.dataset.info).id;
+        //now for all follow action buttons
+        document.querySelectorAll('.follow-action-button').forEach(function(fab){
+            try{
+                //lets check they have info data attributes
+                if( fab.dataset.info ){
+                    //and attempt to parse those, to grab that buttons info user id
+                    let fabUserId = JSON.parse(fab.dataset.infos).id;
+                    //now does that user id match our event buttons user id?
+                    if ( fabUserId && fabUserId === evFabUserId ) {
+                        //yes - time to assign the same state!
+                        assignState(fab, butt.dataset.verb);
```

Comment:
Assuming there are two follow buttons for the identical user on the page. If the first follow button is clicked, `butt.dataset.verb` get modified. ie the verb change from "follow" to "unfollow", which would produce no changes on the second button. `butt.dataset.verb` should be stored in a variable before it is used as a comparison.

Reply:
Ah yup, got you! Sorry, think I got too excited to contribute! 😅 All sorted and updated pushed.